File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/configurations/spring_boot/SpringBootConfigurationSettings.java
package com.github.alexandrecarlton.idea.settings.layout.configurations.spring_boot;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import org.immutables.value.Value;

import java.util.Optional;

@Value.Immutable
@JsonDeserialize(as = ImmutableSpringBootConfigurationSettings.class)
public interface SpringBootConfigurationSettings {
  String mainClass();

  Optional<SpringBootConfigurationEnvironmentSettings> environment();

  Optional<SpringBootConfigurationSpringBootSettings> springBoot();

}

File: layout/src/main/kotlin/com/github/alexandrecarlton/idea/settings/dialog/configurations/spring_boot/SpringBootConfigurationSettings.kt
package com.github.alexandrecarlton.idea.settings.dialog.configurations.spring_boot

import com.fasterxml.jackson.annotation.JsonProperty

data class SpringBootConfigurationSettings(
    @JsonProperty("Main class")
    val mainClass: String,

    @JsonProperty("Environment")
    val environment: SpringBootConfigurationEnvironmentSettings? = null,

    @JsonProperty("Spring Boot")
    val springBoot: SpringBootConfigurationSpringBootSettings? = null
)
